lawnchair/src/app/lawnchair/ui/preferences/data/liveinfo/LiveInformationRequest.kt

```diff
@@ -0,0 +1,49 @@
+package app.lawnchair.ui.preferences.data.liveinfo
+
+import android.util.Log
+import app.lawnchair.ui.preferences.data.liveinfo.model.Announcement
+import app.lawnchair.ui.preferences.data.liveinfo.model.LiveInformation
+import app.lawnchair.util.kotlinxJson
+import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.ResponseBody
+import org.json.JSONObject
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.create
+
+private val retrofit = Retrofit.Builder()
+    .baseUrl("https://lawnchair.app/")
+    .addConverterFactory(kotlinxJson.asConverterFactory("application/json".toMediaType()))
+    .build()
+
+val liveInformationService: LiveInformationService = retrofit.create()
+
+suspend fun getLiveInformation(): LiveInformation? = withContext(Dispatchers.IO) {
+    try {
+        val response: Response<ResponseBody> = liveInformationService.getLiveInformation()
+
+        if (response.isSuccessful) {
+            val responseBody = response.body()?.string() ?: return@withContext null
+
+            val liveInformationObject = JSONObject(responseBody)
+
+            val liveInformation = LiveInformation(
+                announcements = Announcement.fromJsonArray(
+                    liveInformationObject.getJSONArray("announcements"),
+                ),
+            )
```

Comment:
Why are we parsing the JSON manually? Can we use a Retrofit converter instead?

Reply:
Because I thought we don't use them. This code is inspired by the search result code by @MrSluffy. I did check if we use Moshi anywhere in the app though.

I would use Moshi but it is not used in Lawnchair anywhere else so far. As you mentioned on your other comment we should probably use `kotlinx.serialization` instead of this. I will look into updating both requests.

Reply:
We made an exception due to differences of the data response, in startpage the response is like this `["popcorn", [ "popcorn", "popcorn instrumental" ] ]` as we don't want to serialize and map it afterwards, so we use a straightforward approach.

Reply:
Fixed now!